src/components/input/input.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Grow Elastic TextArea
+ *
+ * Increases the `rows` attribute of a `textarea` until it no longer scrolls.
+ *
+ * @param {HTMLTextAreaElement} textarea - the target `textarea` element
+ */
+const growElasticTextArea = (textarea: HTMLTextAreaElement) => {
+  const minRows = Number(textarea.dataset.minRows);
```

Comment:
I wonder if we should make sure this never is below 1?

const minRows = Math.min(Number(textarea.dataset.minRows), 1);

---

What happens if this is not defined on the dataset? Does it fail gracefully?

Reply:
Good idea, @Paul-Hebert!

Reply:
> What happens if this is not defined on the dataset? Does it fail gracefully?

Sounds like a great candidate for a Jest test! 😄

Reply:
@gerardo-rodriguez I set up testing with cypress here: https://github.com/cloudfour/cloudfour.com-patterns/pull/726

I don't think that Jest is the best fit for this case since JSDOM doesn't have a layout engine, so you won't really be able to test out this component. Currently working on adding tests for the textarea component, which I will push to that branch once this PR is merged

But yeah, good point, I will test with no minRows attribute

Reply:
`minRows` will always be set to either the user-provided `rows` attribute or `2`.

If the user sets `rows` to 1, that's a valid value.

If the user sets `rows` to 0 or a negative value, the browser ignores it and the effective result is `2`.

(unfortunately, changing the `rows` value in the Story knob seems to break our JS, so this is difficult to test, but I just verified it)